src/main/java/org/opentripplanner/routing/graph/kryosupport/KryoBuilder.java

```diff
@@ -52,44 +48,28 @@ public static Kryo create() {
         kryo.register(TIntIntHashMap.class, new TIntIntHashMapSerializer());
 
         // Add support for the package local java.util.ImmutableCollections.
-        // Not supported in the current com.conveyal:kryo-tools:1.3.0.
+        // Not supported properly in the current com.conveyal:kryo-tools:1.4.0.
         // This provide support for List.of, Set.of, Map.of and Collectors.toUnmodifiable(Set|List|Map)
         kryo.register(List.of().getClass(), new JavaImmutableListSerializer());
```

Comment:
Doesn't Kryo 5 have built-in support for List.of()? https://github.com/EsotericSoftware/kryo/blob/master/src/com/esotericsoftware/kryo/serializers/ImmutableCollectionsSerializers.java

Or are they broken in the same was as the map one?

Reply:
Yes, they will return a wrong type when serializing an empty Set/List/Map, thus failing the tests which compare the loaded graph. I'll create an issue in kryo about that

Reply:
https://github.com/EsotericSoftware/kryo/blob/master/src/com/esotericsoftware/kryo/serializers/ImmutableCollectionsSerializers.java#L51-L53

and 

https://github.com/EsotericSoftware/kryo/blob/master/src/com/esotericsoftware/kryo/serializers/CollectionSerializer.java#L216-L219

Reply:
Ok, thanks for the explanation. Can you cc me when you create the ticket?